packages/react/src/components/MapCard/Legend.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import { ChevronLeft32, ChevronRight32 } from '@carbon/icons-react';
+
+import { settings } from '../../constants/Settings';
+import Button from '../Button/Button';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  /** true if the legend should be full bleed  and occupy the full width of the map  */
+  isFullWidth: PropTypes.bool.isRequired,
+  /** true if the legend is collapsed (only relevant for isFullWidth:true) */
+  isCollapsed: PropTypes.bool.isRequired,
+  /** true if there should be extra wide margins, like when a card is expanded to full page */
+  increasedMargin: PropTypes.bool,
+  titleText: PropTypes.string,
+  hideLegendText: PropTypes.string,
+  showLegendText: PropTypes.string,
+  /** list of text - color pairs for the legend. Each pair is an array where pos 0 is the text and pos 1 the color */
+  stops: PropTypes.arrayOf(PropTypes.array).isRequired,
+  /** callback for when the collapse toggle button is clicked */
+  onCollapsToggle: PropTypes.func.isRequired,
+  testId: PropTypes.string,
+};
+
+const defaultProps = {
+  titleText: 'Legend',
+  hideLegendText: 'Hide legend',
+  showLegendText: 'Show legend',
+  increasedMargin: false,
+  testId: 'map-legend',
+};
+
+const Legend = ({
+  increasedMargin,
+  isFullWidth,
+  stops,
+  titleText,
+  hideLegendText,
+  showLegendText,
+  isCollapsed,
+  onCollapsToggle,
+  testId,
+}) => {
+  const renderLegendKeys = (stop, i) => {
+    return (
+      <div key={i} className={`${iotPrefix}--map-legend-keys`}>
```

Comment:
Using indices as the key can have potential bugs. It would be better to use one or both of the stop values 
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
Good point!